### ISSUE TITLE
ClientCompatPatch: IndexError

### ISSUE THREAD

**Issue by jmgirven** [ping/instagram_private_api#20]
### Before submitting an issue, make sure you have:
- [x] Read the [README](https://github.com/ping/instagram_private_api/blob/master/README.md) and [docs](https://instagram-private-api.readthedocs.io/en/latest/)
- [x] [Searched](https://github.com/ping/instagram_private_api/search?type=Issues) the bugtracker for similar issues including **closed** ones
- [x] Reviewed the sample code in [tests](https://github.com/ping/instagram_private_api/tree/master/tests) and [examples](https://github.com/ping/instagram_private_api/tree/master/examples)

### Which client are you using?
- [ ] app (``instagram_private_api/``)
- [x] web (``instagram_web_api/``)

### Purpose of your issue?
- [x] Bug report (encountered problems/errors)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### For a *bug report*, you **must** include the Python version used, *code* that will reproduce the error, and the *error log/traceback*.

Paste the output of ``python -V`` here: Python 3.4.6

---

### Describe your issue

This post:

https://www.instagram.com/p/BVRqQxmj2TA/

Has the following node information:

```
node: {
    comments_disabled: false,
    dimensions: {
        height: 1080,
        width: 1080
    },
    display_url: "https://scontent-lht6-1.cdninstagram.com/t51.2885-15/e35/19120472_781130132047194_8491754349737803776_n.jpg",
    edge_liked_by: {
        count: 42
    },
    edge_media_to_caption: {
        edges: [ ]
    },
    edge_media_to_comment: {
        count: 4
    },
    id: "1536194818635424960",
    is_video: false,
    owner: {
        id: "1947139030"
    },
    shortcode: "BVRqQxmj2TA",
    taken_at_timestamp: 1497348714,
    thumbnail_src: "https://scontent-lht6-1.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/19120472_781130132047194_8491754349737803776_n.jpg"
}
```

Notice the `edge_media_to_caption` key returns an empty list. This causes `ClientCompatPatch` to fail with an `IndexError: list index out of range` at https://github.com/ping/instagram_private_api/blob/master/instagram_web_api/compatpatch.py#L52 .

**Comment by jmgirven** [comment]
You could replace that [line](https://github.com/ping/instagram_private_api/blob/master/instagram_web_api/compatpatch.py#L52) with something like:
```
if media.get('caption'):
    caption = media.get('caption')
elif media.get('edge_media_to_caption', {}).get('edges'):
    caption = media.get('edge_media_to_caption', {}).get('edges', [{}])[0].get('node', {}).get('text')
else:
    caption = None
```

**Comment by ping** [comment]
Thanks for the report. Fixed in 9b202042360ebe93a6599c18626d199dc3594209.
